webapp/views.py

```diff
@@ -590,24 +594,27 @@ def post_advantage_subscriptions(preview):
     account_id = payload.get("account_id")
     previous_purchase_id = payload.get("previous_purchase_id")
 
-    # Get the latests active subscription for the account that is making the
-    # purchase, there might be a better endpoint for this in the future.
-    try:
-        subscriptions = advantage.get_account_subscriptions_for_marketplace(
-            account_id=account_id, marketplace="canonical-ua"
-        )
-        last_subscription = subscriptions["subscriptions"][0]
-    except HTTPError:
-        flask.current_app.extensions["sentry"].captureException(
-            extra={"payload": payload}
-        )
-        return (
-            flask.jsonify(
-                {"error": "could not retrieve account subscriptions"}
-            ),
-            500,
-        )
-    except KeyError:
+    if not guest_token:
```

Comment:
What do you think about structuting this as follows
```python
last_subscription = {}
if not guest_token:
    try:
        subscriptions = ...
    except HTTPError:
        ...
    if subscriptions:
        last_subscription = subscriptions[0]
```
and dropping the else block?